src/utils.rs

```diff
@@ -0,0 +1,316 @@
+#[cfg(target_os = "linux")]
+use crate::netlink;
+use crate::{check_command_output_status, Peer, WireguardInterfaceError};
+use std::{collections::HashSet, process::Command};
+
+/// Add peer routing basically copy of wg-quick
+/// On linux system sysctl command is requried to work if using 0.0.0.0/0 or ::/0
+/// For every allowed ip it runs ip `ip_version` route add `allowed_ip` dev `ifname`
+/// For 0.0.0.0/0 allowed ip it runs in order:
+/// ip -4 route add 0.0.0.0/0 dev `ifname` table `fwmark` fwmark is host.fwmark
+/// or default 51820 if value is None
+/// ip -4 rule add not fwmark `host.fwmark` table `host.fwmark`
+/// ip -4 rule add table main supress_prefixlength 0
+/// sysctl -q net.ipv4.conf.all.src_valid_mark=1
+/// iptables-restore -n
+
+#[cfg(target_os = "linux")]
+pub(crate) fn add_peers_routing(
+    peers: &[Peer],
+    ifname: &str,
+) -> Result<(), WireguardInterfaceError> {
+    debug!("Adding peers routing for interface: {}", ifname);
+
+    let mut unique_allowed_ips = HashSet::new();
+    for peer in peers {
+        for addr in &peer.allowed_ips {
+            unique_allowed_ips.insert(addr.to_string());
+        }
+    }
+
+    for allowed_ip in unique_allowed_ips {
+        debug!("Processing allowed IP: {}", allowed_ip);
```

Comment:
Żeby nie powtarzać, to we wszystkich logach jak masz pod ręką zmienną możesz ją wstawić prosto w string tak jak wyżej zrobiłem.